### PR TITLE
Recorder initial

### PR DESCRIPTION
**Pull request by mb-dbc-dk**
Inital version with
kafka/file/stdin input
file/stdout output

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        buildDiscarder(logRotator(artifactDaysToKeepStr: "", artifactNumToKeepStr: "", daysToKeepStr: "30", numToKeepStr: "30"))
+        timestamps()
+    }
+    stages {
+        stage("build") {
+            steps {
+                // Fail Early..
+                script {
+                    if (! env.BRANCH_NAME) {
+                        currentBuild.rawBuild.result = Result.ABORTED
```
Comment (mran-dbc) [inline Jenkinsfile]: currentBuild.rawBuild.result fails if the job is suspended.
currentBuild.result = Result.ABORTED instead.

File: recorder/src/main/java/dk/dbc/solr/performance/recorder/HeaderOutput.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    private static final ObjectMapper O = new ObjectMapper();
+
+    private final Config config;
+
+    public HeaderOutput(Config config) {
+        this.config = config;
+    }
+
+    @Override
+    public void accept(OutputStream os, LogLine logLine) {
```
Comment (mran-dbc) [inline recorder/src/main/java/dk/dbc/solr/performance/recorder/HeaderOutput.java:47]: The parameter logline is only used to get the started timestamp.
Instead use the TS directly and avoid the coupling.

File: recorder/src/main/java/dk/dbc/solr/performance/recorder/OutputWriter.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+     * Stash a log line in the cache, when cache reaches it's limit, output
+     * oldest log line.
+     *
+     * @param logLine log line
+     */
+    @Override
+    public void accept(LogLine logLine) {
+        if (entries.isEmpty())
+            this.origin = logLine.getInstant();
+        long currentOffset = logLine.timeOffsetMS(origin);
```
Comment (mran-dbc) [inline recorder/src/main/java/dk/dbc/solr/performance/recorder/OutputWriter.java:114]: Won't this set the origin to whatever the current timestamp is every time the buffer is emptied, thus resetting the currentOffset back to 0?

**Review by mran-dbc — COMMENTED** [review]
LineSource.java og LinesInputStream.java is usable in both recorder and replayer.
Could they be moved to a common artifact?

File: recorder/src/main/java/dk/dbc/solr/performance/recorder/HeaderOutput.java (hunk 2 above)
Comment (mb-dbc-dk) [reply recorder/src/main/java/dk/dbc/solr/performance/recorder/HeaderOutput.java:47]: Then origin of the timestamp isn't clear

File: recorder/src/main/java/dk/dbc/solr/performance/recorder/OutputWriter.java (hunk 3 above)
Comment (mb-dbc-dk) [reply recorder/src/main/java/dk/dbc/solr/performance/recorder/OutputWriter.java:114]: Entries can only be empty 1st time this is called. But if(origin == null) might be more readable

Comment (mran-dbc) [reply recorder/src/main/java/dk/dbc/solr/performance/recorder/OutputWriter.java:114]: yes - Didn't se that buffer is only emptied during close

**Pull request merged**